src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/IdentifiersShouldHaveCorrectSuffix.cs

```diff
@@ -19,197 +20,202 @@ namespace Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines
-        // The bool values are as mentioned in the Uri
-        private static readonly List<(string typeName, string suffix, bool canSuffixBeCollection)> s_baseTypesAndTheirSuffix = new List<(string, string, bool)>()
-                                                    {
-                                                        ("System.Attribute", "Attribute", false),
-                                                        ("System.EventArgs", "EventArgs", false),
-                                                        ("System.Exception", "Exception", false),
-                                                        ("System.Collections.ICollection", "Collection", false),
-                                                        ("System.Collections.IDictionary", "Dictionary", false),
-                                                        ("System.Collections.IEnumerable", "Collection", false),
-                                                        ("System.Collections.Queue", "Queue", true),
-                                                        ("System.Collections.Stack", "Stack", true),
-                                                        ("System.Collections.Generic.Queue`1", "Queue", true),
-                                                        ("System.Collections.Generic.Stack`1", "Stack", true),
-                                                        ("System.Collections.Generic.ICollection`1", "Collection", false),
-                                                        ("System.Collections.Generic.IDictionary`2", "Dictionary", false),
-                                                        ("System.Collections.Generic.IReadOnlyDictionary`2", "Dictionary", false),
-                                                        ("System.Data.DataSet", "DataSet", false),
-                                                        ("System.Data.DataTable", "DataTable", true),
-                                                        ("System.IO.Stream", "Stream", false),
-                                                        ("System.Security.IPermission","Permission", false),
-                                                        ("System.Security.Policy.IMembershipCondition", "Condition", false)
-                                                    };
-
-        public override void Initialize(AnalysisContext analysisContext)
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(OneSuffixRule, MultipleSuffixesRule);
+
+        // Define for each interface/type the collection of allowed suffixes (first item is the preferred suffix),
+        // note that the order matters as the algorithm works on a first match basis.
+        private static readonly ImmutableArray<(string typeName, ImmutableArray<string> suffixes)> s_baseTypesAndTheirSuffix =
```

Comment:
The other solution I had in mind was to split the `ImmutableArray<string> suffixes` into `string preferredSuffix` and `ImmutableArray<string> additionalSuffixes`. Any preference?